astroid/rebuilder.py

```diff
@@ -133,6 +140,45 @@ def _get_context(
     ) -> Context:
         return self._parser_module.context_classes.get(type(node.ctx), Context.Load)
 
+    def _create_name_node(
+        self,
+        node: Union["ast.ClassDef", "ast.FunctionDef", "ast.AsyncFunctionDef"],
+        parent: Union[nodes.ClassDef, nodes.FunctionDef, nodes.AsyncFunctionDef],
+    ) -> Optional[nodes.AssignName]:
+        if not self._data:
+            return None
+        end_lineno: Optional[int] = getattr(node, "end_lineno", None)
+        if node.body:
```

Comment:
`node` here is the `ast` one. So that should work. AFAIK each function / class body needs at least one node to be valid Python (either a docstring, pass, or something else). In any case if `body` should somehow be empty, it would fallback to the `end_lineno` of the node itself. Or if that also didn't work `None`, which is essentially `self._data[node.lineno - 1 : ]`.